jskit/encoders/bi.py

```diff
@@ -30,184 +31,249 @@ def set_seed(seed):
-    if training_parameters:
-        config['TRAIN_PARAMETERS'].update(training_parameters)
-    if model_parameters:
-        config['MODEL_PARAMETERS'].update(model_parameters)
-        save_restart = True
-    with open("utils/config.cfg", 'w') as configfile:
-        config.write(configfile)
+@ jaseci_action(act_group=['bi_enc'], allow_remote=True)
+def set_train_config(training_parameters: Dict = None):
+    global train_config
+    try:
+        with open("utils/train_config.json", "w+") as jsonfile:
+            train_config.update(training_parameters)
+            json.dump(train_config, jsonfile, indent=4)
+        return "Config setup is complete."
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+
+@ jaseci_action(act_group=['bi_enc'], allow_remote=True)
+def get_model_config():
+    try:
+        with open("utils/model_config.json", "r") as jsonfile:
+            data = json.load(jsonfile)
+            print("Read successful")
+        return data
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+
+@ jaseci_action(act_group=['bi_enc'], allow_remote=True)
+def set_model_config(model_parameters: Dict = None):
+    global model_config
     try:
-        config_setup()
+        with open("utils/model_config.json", "w+") as jsonfile:
+            model_config.update(model_parameters)
+            json.dump(model_config, jsonfile, indent=4)
+
+        config_setup(save_restart=True)
+        return "Config setup is complete."
+
     except Exception as e:
         raise HTTPException(status_code=500, detail=str(e))
-    return "Config setup is complete."
 
 
-@jaseci_action(act_group=['bi_enc'], allow_remote=True)
+@ jaseci_action(act_group=['bi_enc'], allow_remote=True)
 def save_model(model_path: str):
     """
     saves the model to the provided model_path
     """
-    global model, tokenizer, shared, config
+    # global model, tokenizer, model_config
     try:
-        if not model_path.isalnum():
+        if not model_path.replace('_', '').isalnum():
             raise HTTPException(
                 status_code=400,
-                detail='Invalid model name. Only alphanumeric chars allowed.'
+                detail="Invalid model name. Only alphanumeric chars allowed "
             )
         if not os.path.exists(model_path):
             os.makedirs(model_path)
-        if shared == "True":
+        if model_config["shared"] is True:
             model.cont_bert.save_pretrained(model_path)
             tokenizer.save_vocabulary(model_path)
-            with open(model_path+"/config.cfg", 'w') as fp:
-                config.write(fp)
         else:
-            cand_bert_path = os.path.join(model_path)+"/cand_bert/"
-            cont_bert_path = os.path.join(model_path)+"/cont_bert/"
+            cand_bert_path = os.path.join(model_path + "/cand_bert")
+            cont_bert_path = os.path.join(model_path + "/cont_bert")
```

Comment:
should be `os.path.join(model_path, "cand_bert")` right?

Reply:
I tried but doesn't work, somehow